pkg/backend/updates.go

```diff
@@ -27,6 +27,9 @@ type UpdateMetadata struct {
 	// Environment contains optional data from the deploying environment. e.g. the current
 	// source code control commit information.
 	Environment map[string]string `json:"environment"`
+
+	// List of ESC Environments imported as part of the update.
+	EnvironmentImports []string `json:"environmentImports"`
```

Comment:
Feels like this should maybe be in `Environment` as another piece of metadata.

Reply:
I just spoke with @dschaller about this and we will rename `EnvironmentImports` to `EscEnvironments` due to the confusion with `Environment` above. @pgavlin the `Environment` there is completely unrelated to ESC, does it still make sense to put ESC environments inside there?

Reply:
I think so. Per the doc comment on `Environment`:

> Environment contains optional data from the deploying environment. e.g. the current source code control commit information.

To me, the ESC environment(s) in use feels like the same sort of info.

Reply:
(cc @komalali for input here as well)

Reply:
Yeah, I agree with @pgavlin - lets put it in Environment

Reply:
@pgavlin @komalali i think conceptually that makes sense. since we're planning on storing ESC environments as structs (we want to store name and (eventually) resolved revision), is the play here to change `Environments` to be a `map[string]interface{}`? that comes with some annoying type checks, but if we want to stay backwards compatible I think that's the easiest?

Reply:
Isn't the environment here the execution environment? I feel like the esc environment is more configuration and less where it's running

Reply:
I see two options:

1. Change this to a `map[string]any`. Requires changes in the service to accept the new type, but simplifies usage.
2. JSON-encode the list of environments and store that as the value for whatever key we use.

(1) is more work, but is probably preferable.